src/friendly-iframe-embed.js

```diff
@@ -758,84 +785,103 @@ export class FriendlyIframeEmbed {
     const parentWin = toWin(childWin.frameElement.ownerDocument.defaultView);
     setParentWindow(childWin, parentWin);
 
-    // Install necessary polyfills.
-    installPolyfillsInChildWindow(parentWin, childWin);
-
-    // Install runtime styles.
-    installStylesLegacy(
-      childWin.document,
-      ampSharedCss,
-      /* callback */ null,
-      /* opt_isRuntimeCss */ true,
-      /* opt_ext */ 'amp-runtime'
-    );
-
-    // Run pre-install callback.
-    if (opt_preinstallCallback) {
-      opt_preinstallCallback(childWin);
-    }
-
-    // Install embeddable standard services.
-    installStandardServicesInEmbed(childWin);
-
-    // Install built-ins and legacy elements.
-    copyBuiltinElementsToChildWindow(topWin, childWin);
-    stubLegacyElements(childWin);
-
-    const promises = [];
     extensionIds.forEach((extensionId) => {
       // This will extend automatic upgrade of custom elements from top
       // window to the child window.
       if (!LEGACY_ELEMENTS.includes(extensionId)) {
         stubElementIfNotKnown(childWin, extensionId);
       }
+    });
 
-      // Install CSS.
-      const promise = extensions
-        .preloadExtension(extensionId)
-        .then((extension) => {
-          // Adopt embeddable extension services.
-          extension.services.forEach((service) => {
-            installServiceInEmbedIfEmbeddable(childWin, service.serviceClass);
-          });
-
-          // Adopt the custom elements.
-          let elementPromises = null;
-          for (const elementName in extension.elements) {
-            const elementDef = extension.elements[elementName];
-            const elementPromise = new Promise((resolve) => {
-              if (elementDef.css) {
-                installStylesLegacy(
-                  childWin.document,
-                  elementDef.css,
-                  /* completeCallback */ resolve,
-                  /* isRuntime */ false,
-                  extensionId
+    const maybeDelay = getDelayPromise(parentWin);
+    return Promise.resolve()
+      .then(maybeDelay)
+      .then(() => {
+        // Install necessary polyfills.
+        installPolyfillsInChildWindow(parentWin, childWin);
```

Comment:
IMHO, I'm not sure this break down makes sense. Seems too arbitrary to me. A lot of these are just faster to get in immediately. Do we have any data on this?

Reply:
Yeah that seems quite excessive. I have tried to see whether any one of these blocks can be combined. Most (if not all) of these blocks run very fast, but the thing is that they carry their own microtasks, which all contribute to a single long task.

Reply:
So, we already have chunking APIs. Maybe they'd help? I believe they are build on `requestIdleCallback` which allow chunking based on the actual CPU cost.

Reply:
I have looked into the chunk service. Since `requestIdleCallback` is more aggressive than `setTimeout`, I would like to try with this less aggressive version and collect data wrt. its monetary impact first, and then decide which way to go. WDYT?

Reply:
By "aggressive", do you mean it will throttle you too much? I think chunking service allows you to be as "pushy" as you need to be - you are allowed to bust the budget for certain. Also, when the next macrotask will be scheduled is also not given. It certainly won't be `1ms` :). Do we use chunking service for anything else in the inabox?

Reply:
Yes. Make sense. I can try using the highest priority first. Chunking is disabled in inabox because it has larger monetary impact (presumably because the underlying page is less cooperative).

Reply:
I think a similar question still exists, do we want to combine any of these, or you think its better if they are all chunks?